benchexec/test_core_assignment.py

```diff
@@ -100,601 +186,319 @@ def test_singleThread(self):
 
     def test_threeCoresPerRun(self):
         # test all possible numOfThread values for runs with three cores
-        maxThreads = self.cpus * (self.cores // 3)
-        if not self.use_ht and self.ht:
-            maxThreads = self.cpus * (self.cores // 6)
-            if maxThreads == 0:
-                # Test for runs that are split over cpus
-                cpus_per_run = int(math.ceil(3 / (self.cores // 2)))
-                maxThreads = self.cpus // cpus_per_run
-
-        self.assertInvalid(3, maxThreads + 1)
-        if not self.threeCore_assignment:
-            self.skipTest("Need result specified")
-        for num_of_threads in range(1, maxThreads + 1):
-            self.assertValid(
-                3, num_of_threads, self.threeCore_assignment[:num_of_threads]
-            )
+        self.t_unit_assertValid(3, self.threeCore_assignment)
 
     def test_fourCoresPerRun(self):
         # test all possible numOfThread values for runs with four cores
-        maxThreads = self.cpus * (self.cores // 4)
-        if not self.use_ht and self.ht:
-            maxThreads = self.cpus * (self.cores // 8)
-            if maxThreads == 0:
-                # Test for runs that are split over cpus
-                cpus_per_run = int(math.ceil(4 / (self.cores // 2)))
-                maxThreads = self.cpus // cpus_per_run
-
-        self.assertInvalid(4, maxThreads + 1)
-        if not self.fourCore_assignment:
-            self.skipTest("Need result specified")
-        for num_of_threads in range(1, maxThreads + 1):
-            self.assertValid(
-                4, num_of_threads, self.fourCore_assignment[:num_of_threads]
-            )
+        self.t_unit_assertValid(4, self.fourCore_assignment)
 
     def test_eightCoresPerRun(self):
         # test all possible numOfThread values for runs with eight cores
-        maxThreads = self.cpus * (self.cores // 8)
-        if not self.use_ht and self.ht:
-            maxThreads = (self.cpus * self.cores) // 16
-            if maxThreads == 0:
-                # Test for runs that are split over cpus
-                cpus_per_run = int(math.ceil(8 / (self.cores // 2)))
-                maxThreads = self.cpus // cpus_per_run
-        if not maxThreads:
-            self.skipTest(
-                "Testing for runs that need to be split across CPUs is not implemented"
-            )
-        self.assertInvalid(8, maxThreads + 1)
-        if not self.eightCore_assignment:
-            self.skipTest("Need result specified")
-        for num_of_threads in range(1, maxThreads + 1):
-            self.assertValid(
-                8, num_of_threads, self.eightCore_assignment[:num_of_threads]
-            )
+        self.t_unit_assertValid(8, self.eightCore_assignment)
 
 
-class TestCpuCoresPerRun_singleCPU(TestCpuCoresPerRun):
-    cpus = 1
-    cores = 8
-    ht = False
+class Test_Topology_P1_NUMA2_L8_C16_F(TestCpuCoresPerRun):
+    num_of_cores = 16
+    num_of_packages = 1
```

Comment:
The algorithm should be able to handle levels being non-existent, so we test that in some classes. It doesnt make sense to test all variation, that would make too many tests. 
A constant of 1 doesn't mean that a level is unused - we test the code with 1 and 2 packages, since with 2 packages, the code has to create a root level. 

Which core assignments should be identical?